amqp-impl/src/main/java/io/streamnative/pulsar/handlers/amqp/ExchangeContainer.java

```diff
@@ -83,7 +95,14 @@ public CompletableFuture<AmqpExchange> asyncGetExchange(NamespaceName namespaceN
             return existingAmqpExchangeFuture;
         } else {
             String topicName = PersistentExchange.getExchangeTopicName(namespaceName, exchangeName);
-            CompletableFuture<Topic> topicCompletableFuture = amqpTopicManager.getTopic(topicName, createIfMissing);
+
+            Map<String, String> newProperties = null;
+            if (createIfMissing && arguments != null) {
```

Comment:
Maybe we can serialize declare custom properties to a JSON string, and put it to a specific key.